snuba/migrations/runner.py

```diff
@@ -61,10 +60,7 @@ def show_all(self) -> List[Tuple[MigrationGroup, List[MigrationStatus]]]:
         """
         migrations: List[Tuple[MigrationGroup, List[MigrationStatus]]] = []
 
-        migration_status = self._get_migration_status()
-
-        def get_status(migration_key: MigrationKey) -> Status:
-            return migration_status.get(migration_key, Status.NOT_STARTED)
+        get_status = self._get_migration_status_func()
```

Comment:
I think the previous approach was more clear. It made clear that the status is fetched only once for this method and the nested function is used as a utility method to access those results. 
With this approach instead the question arises why cannot we just load the status for a given migration id instead of getting the function first. 

If you do not want to duplicate the code of the `get_status` functions, why not creating a `MigrationGroupStatus` dataclass with a factory method that loads data from the DB and a `get_status` method that returns the status for a given migration ID ?